### DISCUSSION TITLE
Using Context Key Service

### DISCUSSION THREAD

**Discussion by maheshrajrp** [eclipse-theia/theia#11372] [Q&A] [answered]
Hi,
In our extension we have CustomEditor. How shall I create Context for the same and use it in when clause like `isCustomEditorFocused`. I believe we need to implement a ContextKeyService like `ContextKeyServiceDummyImpl` for our extension. However, am unable to locate a better example. The monaco example seems a bit off. Do we have any simple example ? Also, Am I going in the right track or missing something. 

Thanks

**Comment by colin-grant-work** [comment, marked as answer]
@maheshrajrp, if all you need to do is add a context key, you don't need to implement a whole `ContextKeyService`. A good example that is nearly parallel to your case is the [`ResourceContextKey`](https://github.com/eclipse-theia/theia/blob/master/packages/core/src/browser/resource-context-key.ts). You just need to register your new context key and hook into appropriate events (for example `ApplicationShell.onDidChangeActiveWidget`) to update the value of the context key when the context value changes.

**Reply by maheshrajrp** [reply]
Hi @colin-grant-work ,
Thanks for Responding. This answer suits for most of my cases. However, in a similar manner, is there any recommended way to tap into Dialogs to get the active dialogs too ?

Thanks

**Reply by colin-grant-work** [reply]
You may be able to get what you need from the [`DialogOverlayService`](https://github.com/eclipse-theia/theia/blob/97ce5f89eccf866fdd2058c20b50a8271204790f/packages/core/src/browser/dialogs.ts#L77), which has a record of all of the dialogs active in the application, though I haven't worked with it very much myself.
